skimage/measure/_regionprops.py

```diff
@@ -221,88 +235,89 @@ def mean_intensity(self):
     def min_intensity(self):
         return np.min(self.intensity_image[self.image])
 
-    @only2d
     def major_axis_length(self):
-        l1, _ = self.inertia_tensor_eigvals
+        l1 = self.inertia_tensor_eigvals[0]
         return 4 * sqrt(l1)
 
-    @only2d
     def minor_axis_length(self):
-        _, l2 = self.inertia_tensor_eigvals
+        l2 = self.inertia_tensor_eigvals[-1]
         return 4 * sqrt(l2)
 
     @_cached
-    @only2d
     def moments(self):
-        return _moments.moments(self.image.astype(np.uint8), 3)
+        M = _moments.moments(self.image.astype(np.uint8), 3)
+        if not self._warned:
+            from warnings import warn
+            warn(XY_TO_RC_DEPRECATION_MESSAGE)
+            self.warned = True
```

Comment:
This is how warnings work by default.  Anyway, not up to us to set this up; warn and let the user worry about filter settings.